PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/FunctionDeploymentPanel.java

```diff
@@ -84,15 +248,118 @@ protected JLabel getLblMavenProject() {
 
     @Override
     protected void resetFromConfig(@NotNull FunctionDeployConfiguration configuration) {
-
+        configuration.setFunctionId(configuration.getFunctionId());
+        final Module previousModule = configuration.getModule();
+        if (previousModule != null) {
+            for (int i = 0; i < cbFunctionModule.getItemCount(); i++) {
+                final Module module = cbFunctionModule.getItemAt(i);
+                if (Paths.get(module.getModuleFilePath()).equals(Paths.get(previousModule.getModuleFilePath()))) {
+                    cbFunctionModule.setSelectedIndex(i);
+                    break;
+                }
+            }
+        }
+        presenter.loadFunctionApps(false);
     }
 
     @Override
     protected void apply(@NotNull FunctionDeployConfiguration configuration) {
+        if (selectedFunctionApp == null || selectedFunctionApp.getResource() == null) {
+            return;
+        }
+        final FunctionApp targetFunction = selectedFunctionApp.getResource();
+        configuration.setAppName(targetFunction.name());
```

Comment:
shall add a method accept FunctionApp  in configuration

Reply:
Done